### PR TITLE
MAGECLOUD-3486: [Cloud Docker] Add validations to improve error handling

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
Removed possibility to provide empty option values like `php bin/ece-tools docker:build --es`
Now services options require some value.
 

### Fixed Issues (if relevant)

https://magento2.atlassian.net/browse/MAGECLOUD-3486

### Manual testing scenarios

1. Run command  `php bin/ece-tools docker:build --es`
2. Command returns an error
```
 The "--es" option requires a value.  
```
3. Repeat with different services.

See [test](https://jira.corp.magento.com/browse/MAGECLOUD-50)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Command/Docker/Build.php [hunk 1]
```diff
@@ -116,32 +116,32 @@ protected function configure()
...
                 self::OPTION_PHP,
                 null,
-                InputOption::VALUE_OPTIONAL,
+                InputOption::VALUE_REQUIRED,
                 'PHP version'
             )->addOption(
                 self::OPTION_NGINX,
                 null,
-                InputOption::VALUE_OPTIONAL,
+                InputOption::VALUE_REQUIRED,
```
Comment (shiftedreality) [inline src/Command/Docker/Build.php:116]: It must be optional because by default we're parsing `.magetno.env.yaml` file

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: src/Command/Docker/Build.php (hunk 1 above)
Comment (oshmyheliuk) [reply src/Command/Docker/Build.php:116]: It's not about option it's about value.
With `InputOption::VALUE_OPTIONAL` you can pass `--nginx` without value.
With `InputOption::VALUE_REQUIRED` you can't pass `--nginx` without value.
An option will still be optional but can't be added without value.

Comment (shiftedreality) [reply src/Command/Docker/Build.php:116]: Got it, thanks for clarification

**Review by shiftedreality — DISMISSED** [review]

**Review by arhiopterecs — DISMISSED** [review]

**Review by NadiyaS — DISMISSED** [review]

**Comment by andriyShevtsov** [comment]
QA passed

File: src/Command/Docker/Build.php [hunk 2]
```diff
@@ -186,26 +198,33 @@ public function execute(InputInterface $input, OutputInterface $output)
...
             }
-        });
+        }
 
-        $versionList = $this->serviceConfig->getAllServiceVersions($config);
+        if ($input->isInteractive()) {
+            $versionList = $this->serviceConfig->getAllServiceVersions($config);
 
-        $unsupportedErrorMsg = $this->versionValidator->validateVersions($versionList);
+            $unsupportedErrorMsg = array_merge($this->versionValidator->validateVersions($versionList));
```
Comment (NadiyaS) [inline src/Command/Docker/Build.php]: array_merge() should be removed

Comment (oshmyheliuk) [reply src/Command/Docker/Build.php]: removed

File: src/Command/Docker/Build.php [hunk 3]
```diff
@@ -233,4 +252,31 @@ public function isEnabled(): bool
...
+     * @param array $emptyVersions
+     * @return string
+     */
+    private function createQuestionMessage(array $unsupportedErrorMsg, array $emptyVersions): string
+    {
+        $questionMessage = '';
+        if ($emptyVersions) {
+            $questionMessage .= sprintf(
+                "Some service versions were not provided:\n" .
+                "Using the versions specified in your configuration files for the following services: %s\n" .
```
Comment (NadiyaS) [inline src/Command/Docker/Build.php]: There are just my thoughts :) : 
 - this message says nothing whether service be installed or not. So, user may just guess or check later.
 - we already know which services are configured in the files ($versionList, line 214) and we may certainly say that something won't be installed.
It's up to you, what you will do (or won't do) after reading this comment :-D

**Review by NadiyaS — APPROVED** [review]

**Review by NadiyaS — DISMISSED** [review]

File: src/Command/Docker/Build.php [hunk 4]
```diff
@@ -233,4 +252,31 @@ public function isEnabled(): bool
...
+                "If the service is missing in configuration files it will not be installed.\n",
```
Comment (hguthrie) [inline src/Command/Docker/Build.php]: "Only services defined in the configuration files will be installed."

Comment (hguthrie) [reply src/Command/Docker/Build.php]: or "If a service is not defined in the configuration files, it will not be installed."

Comment (oshmyheliuk) [reply src/Command/Docker/Build.php]: I already changed to the first variant. Do I need to change the message to the second variant?

Comment (oshmyheliuk) [reply src/Command/Docker/Build.php]: Actually second is better, I'll change

**Review by NadiyaS — DISMISSED** [review]

**Review by shiftedreality — DISMISSED** [review]

**Review by NadiyaS — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**